cfsdk/src/main/java/io/harness/cfsdk/CfClient.java

```diff
@@ -557,6 +557,11 @@ protected void doInitialize(
         this.configuration = configuration;
         this.target = target;
 
+        if (apiKey == null || apiKey.trim().isEmpty()) {
+            SdkCodes.errorMissingSdkKey();
+            throw new IllegalArgumentException("missing sdk key");
```

Comment:
Forgive my Java ignorance - but should we not indicate in the signature the SDK can throw this exception when initializing the client so the user can add a clause for it ? I notice it already informs that it throws for `IllegalStateException`

Reply:
It's an unchecked exception because it inherits from RuntimeException, so there's no need to add an explicit throws to the sig or try/catch to the call.

Reply:
Looks like IllegalStateException is a Runtime too, guess the original author added it to make it more explicit in the signature.

Reply:
👍